.github/workflows/ci.yml

```diff
@@ -17,16 +17,89 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: 'npm'
-      - run: npm ci
+      - uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm
+      - name: Install dependencies if cache invalid
```

Comment:
If we have stuff in cache that matches this hash - use it. if we do not, run clean install.

clean install runs an install from the package-lock so it should copy the exact node modules that were used to build this PR.

npm install - will install and try to pull more recent versions given the `^` or `~` etc.

Reply:
Makes sense.